massa-execution-worker/src/execution.rs

```diff
@@ -1128,16 +1128,17 @@ impl ExecutionState {
         address: &Address,
         key: &[u8],
     ) -> (Option<Vec<u8>>, Option<Vec<u8>>) {
-        let final_entry = self.final_state.read().ledger.get_data_entry(address, key);
+        let binding = self.final_state.read();
+        let final_entry = binding.ledger.get_data_entry(address, key);
         let search_result = self
             .active_history
             .read()
             .fetch_active_history_data_entry(address, key);
         (
-            final_entry.clone(),
+            final_entry.as_ref().map(|res| res.deref().to_vec()),
```

Comment:
I don't understand how it reduce allocations. In my mind, `to_vec()` makes a new allocation for the vector. I think I'm missing something. Can you explain it to me ?

Reply:
At this point in the codebase, the refactor is redundant, true. I'm waiting on @Eitu33 to discuss if replacing (alloc-heavy) `Vec<u8>` with a `Box<dyn Deref<Target = [u8]>>` might actually be useful.  

Assuming that we are all for it, the plan is to put in the change wherever we are cloning out the bytes, when just having a pinned reference into the DB will do.

One other consideration I have in mind, is to establish the pattern that when data is cloned out, it is done so into its deserialized form. Idea being "when you don't own the data, you hold a reference to the bytes that exist in the DB. When you _do_ own the data, you own it in its a cannonical type (e.g. as a byte-vec in a SecureSend, or the struct itself)".

Reply:
That's interesting yes. Maybe be worth looking in stabilization phase